mesa/batchrunner.py

```diff
@@ -118,7 +118,7 @@ def collect_model_vars(self, model):
     def collect_agent_vars(self, model):
         """ Run reporters and collect agent-level variables. """
         agent_vars = {}
-        for agent in model.schedule.agents:
+        for agent in model.schedule.agents.values():
```

Comment:
A sure-fire fix is to used `collections.OrderedDict`. But this will be slower since in Python 3 it is not written in C. But correctness > speed in this case.